src/mbedtls_context.cpp

```diff
@@ -642,19 +642,27 @@ namespace sockpp {
 #elif defined(_WIN32)
     // Windows:
     static string read_system_root_certs() {
-	    PCCERT_CONTEXT pContext = nullptr;
-	    HCERTSTORE hStore = CertOpenSystemStore(NULL, "ROOT");
-		if(hStore == nullptr) {
-			return "";
-		}
-
-		stringstream certs;
-		while ((pContext = CertEnumCertificatesInStore(hStore, pContext))) {
-			certs.write((const char*)pContext->pbCertEncoded, pContext->cbCertEncoded);
-		}
-
-		CertCloseStore(hStore, CERT_CLOSE_STORE_FORCE_FLAG);
-		return certs.str();
+        PCCERT_CONTEXT pContext = nullptr;
+        HCERTSTORE hStore = CertOpenSystemStore(NULL, "ROOT");
+        if(hStore == nullptr) {
+            return "";
+        }
+
+        stringstream certs;
+        while ((pContext = CertEnumCertificatesInStore(hStore, pContext))) {
+            DWORD pCertPEMSize = 0;
+            if (!CryptBinaryToStringA(pContext->pbCertEncoded, pContext->cbCertEncoded, CRYPT_STRING_BASE64HEADER, NULL, &pCertPEMSize)) {
+                return "";
+            }
+            LPSTR pCertPEM = (LPSTR)malloc(pCertPEMSize);
```

Comment:
It looks like this is never freed, and so it would cause a memory leak.

Reply:
Thank you, should be fixed now.